core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -0,0 +1,388 @@
+          sql.setString(4, JdbcUtil.namespaceToString(from.namespace()));
+          sql.setString(5, from.name());
+          return sql.executeUpdate();
+        }
+      });
+
+      if (updatedRecords == 1) {
+        LOG.debug("Successfully renamed table from {} to {}!", from, to);
+      } else if (updatedRecords == 0) {
+        throw new NoSuchTableException("Failed to rename table! Table '%s' not found in the catalog!", from);
+      } else {
+        throw new RuntimeException("Failed to rename table! Rename operation Failed");
+      }
+
+    } catch (SQLIntegrityConstraintViolationException e) {
+      throw new AlreadyExistsException("Table with name '%s' already exists in the catalog!", to);
+    } catch (SQLException e) {
+      throw new UncheckedSQLException(e, "Failed to rename table");
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to rename", e);
+    }
+  }
+
+  @Override
+  public String name() {
+    return catalogName;
+  }
+
+  @Override
+  public Configuration getConf() {
+    return conf;
+  }
+
+  @Override
+  public void setConf(Configuration conf) {
+    this.conf = conf;
+  }
+
+  @Override
+  public void createNamespace(Namespace namespace, Map<String, String> metadata) {
+    throw new UnsupportedOperationException("Cannot create namespace " + namespace +
+        ": createNamespace is not supported");
+  }
+
+  @Override
+  public List<Namespace> listNamespaces(Namespace namespace) throws NoSuchNamespaceException {
+    if (!namespaceExists(namespace)) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+    }
+
+    try {
+
+      List<Namespace> namespaces = connections.run(conn -> {
+        List<Namespace> result = Lists.newArrayList();
+
+        try (PreparedStatement sql = conn.prepareStatement(JdbcUtil.LIST_NAMESPACES_SQL)) {
+          sql.setString(1, catalogName);
+          sql.setString(2, JdbcUtil.namespaceToString(namespace) + "%");
+          ResultSet rs = sql.executeQuery();
+          while (rs.next()) {
+            result.add(JdbcUtil.stringToNamespace(rs.getString(JdbcUtil.TABLE_NAMESPACE)));
+          }
+          rs.close();
+        }
+
+        return result;
+      });
+
+      int subNamespaceLevelLength = namespace.levels().length + 1;
+      namespaces = namespaces.stream()
+          // exclude itself
+          .filter(n -> !n.equals(namespace))
+          // only get sub namespaces/children
+          .filter(n -> n.levels().length >= subNamespaceLevelLength)
+          // only get sub namespaces/children
+          .map(n -> Namespace.of(
+              Arrays.stream(n.levels()).limit(subNamespaceLevelLength).toArray(String[]::new)
+              )
+          )
+          // remove duplicates
+          .distinct()
+          .collect(Collectors.toList());
+
+      LOG.debug("From the namespace '{}' found: {}", namespace, namespaces);
```

Comment:
I prefer not to add logs like this that take up so much space. Debug is okay, but in general I would avoid logs that simply state what happened (and in verbose detail) during normal execution unless it is a significant state change for the DB (like renamed, dropped, or created).